django_extensions/management/commands/sqldsn.py

```diff
@@ -129,23 +127,14 @@ def postgresql(self, dbhost, dbport, dbname, dbuser, dbpass, dsn_style=None):
                                      dbpass))
 
         if dsn_style == 'all' or dsn_style == 'uri':
-            if dbport is not None:
-                dsnstr = "postgresql://{3}:{4}@{0}:{1}/{2}"
-            else:
-                dsnstr = "postgresql://{3}:{4}@{0}/{2}"
+            dsnstr = "postgresql://{user}:{password}@{host}/{name}"
 
-            dsn.append(dsnstr.format(dbhost,
-                                     dbport,
-                                     dbname,
-                                     dbuser,
-                                     dbpass,))
+            dsn.append(dsnstr.format(
+                host=":".join(map(str, filter(None, [dbhost, dbport]))),
```

Comment:
Could we use: `"{host}:{port}".format(host=dbhost, port=dbport) if dbport else dbhost`
(shorter `"%s:%s" % (dbhost, dbport)` is also fine) instead of the `map` and `filter` combo ?

Which I like for data parsing but imho makes these cases harder to read.

Reply:
@trbs changed, readability counts ;)